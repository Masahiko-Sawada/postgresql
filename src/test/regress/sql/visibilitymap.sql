--
-- Visibility map
--
CREATE TABLE vmtest (i INT primary key);
INSERT INTO vmtest SELECT generate_series(1,10000);

\set VERBOSITY terse

-- All pages are become all-visible
VACUUM vmtest;
SELECT relallvisible  = (pg_relation_size('vmtest') / current_setting('block_size')::int) FROM pg_class WHERE relname = 'vmtest';

-- All pages are become all-frozen
VACUUM FREEZE vmtest;
SELECT relallfrozen  = (pg_relation_size('vmtest') / current_setting('block_size')::int) FROM pg_class WHERE relname = 'vmtest';

-- All pages are skipped according to VM
VACUUM FREEZE VERBOSE vmtest;

DROP TABLE vmtest;
