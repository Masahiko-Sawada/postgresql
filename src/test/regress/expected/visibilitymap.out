--
-- Visibility map
--
CREATE TABLE vmtest (i INT primary key);
INSERT INTO vmtest SELECT generate_series(1,10000);
\set VERBOSITY terse
-- All pages are become all-visible
VACUUM vmtest;
SELECT relallvisible  = (pg_relation_size('vmtest') / current_setting('block_size')::int) FROM pg_class WHERE relname = 'vmtest';
 ?column? 
----------
 t
(1 row)

-- All pages are become all-frozen
VACUUM FREEZE vmtest;
SELECT relallfrozen  = (pg_relation_size('vmtest') / current_setting('block_size')::int) FROM pg_class WHERE relname = 'vmtest';
 ?column? 
----------
 t
(1 row)

-- All pages are skipped according to VM
VACUUM FREEZE VERBOSE vmtest;
INFO:  vacuuming "public.vmtest"
INFO:  index "vmtest_pkey" now contains 10000 row versions in 30 pages
INFO:  skipped 45 frozen pages according to visibility map
INFO:  "vmtest": found 0 removable, 0 nonremovable row versions in 0 out of 45 pages
DROP TABLE vmtest;
